go/parquet/file/file_writer.go

```diff
@@ -30,21 +30,19 @@ import (
 
 // Writer is the primary interface for writing a parquet file
 type Writer struct {
-	sink           utils.WriteCloserTell
-	open           bool
-	props          *parquet.WriterProperties
-	rowGroups      int
-	nrows          int
-	metadata       metadata.FileMetaDataBuilder
-	fileEncryptor  encryption.FileEncryptor
-	rowGroupWriter *rowGroupWriter
+	sink                    utils.WriteCloserTell
+	open                    bool
+	props                   *parquet.WriterProperties
+	rowGroups               int
+	nrows                   int
+	metadata                metadata.FileMetaDataBuilder
+	fileEncryptor           encryption.FileEncryptor
+	rowGroupWriter          *rowGroupWriter
+	initialKeyValueMetadata metadata.KeyValueMetadata
```

Comment:
Since we just initialize the `FileMetaDataBuilder` with the initialKeyValueMetadata, we don't actually need to store it after we initialize things.

We could probably modify the `WriteOption` and how the config works by introducing a config struct like `type config struct { props *parquet.WriterProperties; initialKVMetadata metadata.KeyValueMetadata }`  and have `WriteOption` modify the config struct to initialize the writer. This would let us remove the `initialKeyValueMetadata` member entirely from the `Writer`.

It's not necessary (things work just fine currently and I think this is good) but it's an idea.

Reply:
@zeroshade - I'm happy to make the change to the `WriteOption` interface if you think that makes sense.  I wasn't sure how much of a breaking change would be acceptable.

I could add https://github.com/tschaub/arrow/commit/84cefb58a183fd5e810c6515361c2fbaaf61673e to this branch if that sounds good.

Reply:
I like it, let's add that to this branch

Reply:
Commit added.  It looks like the [failing Dev workflow](https://github.com/apache/arrow/actions/workflows/dev.yml) is unrelated.